evoting/service/service_test.go

```diff
@@ -308,12 +313,296 @@ func TestEvolveRoster(t *testing.T) {
 		Admins:    []uint32{idAdmin, idAdmin2},
 	})
 	require.Nil(t, err)
-	log.Lvl2("Wrote 2nd roster")
 
 	// Run an election on the new set of conodes, the new nodeKP, and the new
 	// election admin.
 	runAnElection(t, s0, rl, nodeKP, idAdmin2)
 
 	// There was a test here before to try to replace the leader.
 	// It didn't work. For the time being, that is not supported.
+
+	// The decrypt protocol tries to stop early as soon as 2n/3 + 1 nodes store a partial.
+	// However, since the leader sends a broadcast to all the n nodes initially we
+	// want the servers to be up until the goroutines terminate or the test framework complains
+	// about zombie goroutines. The call to time.Sleep ensures we dont end up with
+	// zombie goroutines
+	time.Sleep(5 * time.Second)
```

Comment:
I do understand that sometimes `time.Sleep` is the only way to go forward. But most of the time I managed to do something like:

```
for i := 0; i < 10; i++{
  if checkIfAllMessagesAreDone(){
    break
  }
  time.After(200*time.Millisecond):
}
require.NotEqual(t, 10, i, "some messages are still waiting")
```

Can you look whether that is possible? Also for the other occurrences of `time.Sleep` in previous tests. If it's too complicated, leave the `time.Sleep` in.